packages/test-renderer/src/createWebGLContext.ts

```diff
@@ -516,4 +516,4 @@ class WebGLRenderingContext {
   getShaderInfoLog = () => ''
 }
 
-export const createWebGLContext = (canvas: HTMLCanvasElement) => new WebGLRenderingContext(canvas)
+export const createWebGLContext = (canvas: HTMLCanvasElement) => new WebGLRenderingContext(canvas) as any
```

Comment:
is `any` right here? `any` is very broad.

Reply:
`createWebGLContext` doesn't actually implement a WebGL/WebGL 2 context, so it can't later satisfy `canvas#getContext`.

Reply:
Yeah, I had trouble getting this one right. However, the previous version may have caused other problems in typing, like:
```
//@ts-ignore
canvas.getContext = (type: string): string => {
  if (type === 'webgl' || type === 'webgl2') {
    return createWebGLContext(canvas)
  }
}
```
string as returned type is correct, but it shouldn't be.

Reply:
Because the `getContext` is overloaded, I had to overload our implementations as well.
```
  function getContext(
  function getContext(
    contextId: '2d',
    options?: CanvasRenderingContext2DSettings,
  ): CanvasRenderingContext2D | null
  function getContext(
    contextId: 'bitmaprenderer',
    options?: ImageBitmapRenderingContextSettings,
  ): ImageBitmapRenderingContext | null
  function getContext(contextId: 'webgl', options?: WebGLContextAttributes): WebGLRenderingContext | null
  function getContext(contextId: 'webgl2', options?: WebGLContextAttributes): WebGL2RenderingContext | null
  function getContext(contextId: string): RenderingContext | null {
    if (contextId === 'webgl' || contextId === 'webgl2') {
      return createWebGLContext(canvas)
    }
    return null
  }
  
  canvas.getContext = getContext
```
Maybe there's a better solution, but I couldn't find it.